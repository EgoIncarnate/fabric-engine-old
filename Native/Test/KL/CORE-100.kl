struct S
{
  Integer a[];
  Scalar b;
};

function Scalar getB( S s )
{
  return s.b;
}

function entry()
{
  var S s;
  s.a.push( 17 );
  s.b = 3.14;
  report getB( s );
}
