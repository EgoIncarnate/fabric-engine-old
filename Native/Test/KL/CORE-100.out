GlobalList
  StructDecl( 'S' )
    MemberDecl( 'a', Integer[] )
    MemberDecl( 'b', Scalar )
  Function( 'getB' ('getB'), Scalar:r-value )
    ParamList
      Param( 's', S, r-value )
    CompoundStatement
      StatementList
        ReturnStatement
          StructMemberOp( 'b' )
            Var('s')
  Function( 'entry' ('entry'), void )
    ParamList
    CompoundStatement
      StatementList
        StatementList
          VarDecl( 's', S )
        ExprStatement
          MethodOp( 'push' )
            StructMemberOp( 'a' )
              Var('s')
            ArgList
              ConstSize( 17 )
        ExprStatement
          AssignOp(ASSIGN_OP)
            StructMemberOp( 'b' )
              Var('s')
            ConstScalar( 3.14 )
        Report
          Call('getB')
            ArgList
              Var('s')
3.14
